kernel/stack.c

```diff
@@ -56,15 +56,14 @@ void _impl_k_stack_init(struct k_stack *stack, u32_t *buffer, int num_entries)
 }
 
 #ifdef CONFIG_USERSPACE
-u32_t _handler_k_stack_init(u32_t stack, u32_t buffer, u32_t num_entries_p,
-			    u32_t arg4, u32_t arg5, u32_t arg6, void *ssf)
+_SYSCALL_HANDLER3(k_stack_init, stack, buffer, num_entries_p)
 {
 	int num_entries = (int)num_entries_p;
 
 	/* FIXME why is 'num_entries' signed?? */
-	_SYSCALL_VERIFY(num_entries > 0, ssf);
-	_SYSCALL_OBJ_INIT(stack, K_OBJ_STACK, ssf);
-	_SYSCALL_MEMORY_ARRAY_WRITE(buffer, num_entries, sizeof(u32_t), ssf);
+	_SYSCALL_VERIFY(num_entries > 0);
+	_SYSCALL_OBJ_INIT(stack, K_OBJ_STACK);
+	_SYSCALL_MEMORY_ARRAY_WRITE(buffer, num_entries, sizeof(u32_t));
```

Comment:
_SYSCALL_MEMORY_ARRAY_WRITE() will perform the multiplication and overflow check as if num_entries were unsigned.  Can this be a problem?

Reply:
i don't know. I would prefer to look into this in another patch set.

Reply:
Alright.

Reply:
I fixed this in #4302